action.yml

```diff
@@ -22,6 +22,63 @@ inputs:
     description: 'Whether to run scorecard as part of the analysis (no/yes/verbose default to "no")'
     required: false
     default: "no"
+  upload_code_scanning:
+    description: 'Whether to upload the results to GitHub Code Scanning'
+    required: false
+    default: "true"
+  compile_legitify:
+    description: "Compile legitify from source (use legitify_base_version as ref)"
+    required: false
+    default: "false"
 runs:
-  using: "node16"
-  main: "index.js"
+  using: "composite"
+  steps:
+    - uses: actions/setup-node@v3
+      with:
+          node-version: '16'
+
+    - id: setup-go
+      if: ${{ inputs.compile_legitify == 'true' }}
+      uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # ratchet:actions/setup-go@v3
+      with:
+        go-version: 1.19
+    - id: compile_legitify
+      if: ${{ inputs.compile_legitify == 'true' }}
+      shell: bash
+      env:
+        ref: "${{ inputs.legitify_base_version }}"
+      run: |
+        git clone https://github.com/Legit-Labs/legitify.git legitify-build
+        cd legitify-build
+        git checkout "${ref}"
+        go build -o "${GITHUB_ACTION_PATH}/legitify"
+    - id: analyze
+      shell: bash
+      env:
+        github_token: ${{ inputs.github_token }}
+        analyze_self_only: ${{ inputs.analyze_self_only }}
+        repositories: ${{ inputs.repositories }}
+        legitify_base_version: ${{ inputs.legitify_base_version }}
+        scorecard: ${{ inputs.scorecard }}
+        upload_code_scanning: ${{ inputs.upload_code_scanning }}
+        compile_legitify: ${{ inputs.compile_legitify }}
```

Comment:
```suggestion
        pre_installed: ${{ inputs.compile_legitify }}
```

Reply:
Maybe extract the download part from the node action to the composite instead of passing a flag